packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts

```diff
@@ -154,49 +163,32 @@ export class SafeMoneriumClient extends MoneriumClient {
     }
   }
 
+  /**
+   * Get the current chain id
+   * @returns The Chain Id
+   */
+  async getChainId(): Promise<number> {
+    return Number(BigInt(await this.#safeSdk.getChainId()))
+  }
+
   /**
    * Get the corresponding Monerium SDK Chain from the current chain id
    * @returns The Chain
    */
   async getChain(): Promise<Chain> {
-    const chainId = await this.#safeSdk.getChainId()
+    const chainId = BigInt(await this.#safeSdk.getChainId())
 
-    switch (chainId.toString()) {
-      case '1':
-      case '5':
-        return 'ethereum'
-      case '100':
-      case '10200':
-        return 'gnosis'
-      case '137':
-      case '80001':
-        return 'polygon'
-      default:
-        throw new Error(`Chain not supported: ${chainId}`)
-    }
+    return getMoneriumChain(Number(chainId))
   }
 
   /**
    * Get the corresponding Monerium SDK Network from the current chain id
    * @returns The Network
    */
   async getNetwork(): Promise<Networks> {
-    const chainId = await this.#safeSdk.getChainId()
+    const chainId = BigInt(await this.#safeSdk.getChainId())
```

Comment:
It already returns a BigInt

```suggestion
    const chainId = await this.#safeSdk.getChainId()
```